lib/oneview-sdk/resource/api200/logical_interconnect_group.rb

```diff
@@ -53,14 +47,25 @@ def self.get_default_settings(client)
       # @param [String] type Interconnect type
       # @raise [StandardError] if a invalid type is given then raises an error
       def add_interconnect(bay, type)
+        interconnect_type = OneviewSDK::Interconnect.get_type(@client, type)
+        raise OneviewSDK::NotFound unless interconnect_type
+
+        entry_already_present = false
         @data['interconnectMapTemplate']['interconnectMapEntryTemplates'].each do |entry|
           entry['logicalLocation']['locationEntries'].each do |location|
             if location['type'] == 'Bay' && location['relativeValue'] == bay
-              entry['permittedInterconnectTypeUri'] = OneviewSDK::Interconnect.get_type(@client, type)['uri']
+              entry['permittedInterconnectTypeUri'] = interconnect_type['uri']
+              entry_already_present = true
             end
           end
         end
-      rescue StandardError
+
+        unless entry_already_present
+          new_entry = new_interconnect_entry_template(bay, interconnect_type['uri'])
+          @data['interconnectMapTemplate']['interconnectMapEntryTemplates'] << new_entry
+        end
+
+      rescue OneviewSDK::NotFound
         list = OneviewSDK::Interconnect.get_types(@client).map { |t| t['name'] }
         raise "Interconnect type #{type} not found! Supported types: #{list}"
```

Comment:
I'm wondering why raising and rescuing a NotFound error, to then raise a StandardError